include/ocvsmd/sdk/defines.hpp

```diff
@@ -8,13 +8,35 @@
 
 #include <cetl/pf20/cetlpf.hpp>
 
+#include <cerrno>
 #include <cstdint>
 
 namespace ocvsmd
 {
 namespace sdk
 {
 
+/// Defines some common error codes of IPC operations.
+///
+/// Maps to `errno` values, hence `std::int32_t` inheritance and zero on success.
+///
+enum class ErrorCode : std::int32_t  // NOLINT
+{
+    Success             = 0,
```

Comment:
Success is not an error; it should be represented by an absence of an error, like empty `optional<ErrorCode>`. Further, in your existing code there are many occrrences where the ErrorCode can only appear to indicate an error, thus the Success case is redundant and possibly confusing.

Reply:
IMO, although yes there is "Success" redundancy issue, but at least everything is simple, and IMHO with very small amount of confusion. If you insist on `optional<ErrorCode` I would like to try it in a separate PR - currently I'm not sure that going with optional will make code practically better (besides pure academic point), but I might be wrong.

Reply:
It is necessary but we can introduce this change in the next PR.